### PR TITLE
Fix audit logs

### PR DESCRIPTION
**Pull request by bergerhoffer**
Replaces #20856.

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@soltysh I added the fields as a table, would you mind reviewing?

https://pr-20856-update--ocpdocs.netlify.com/openshift-enterprise/latest/nodes/nodes/nodes-nodes-audit-log.html#nodes-pods-audit-log-basic_nodes-nodes-audit-log

**Comment by bergerhoffer** [comment]
@openshift/team-documentation For peer review please.

**Comment by mburke5678** [comment]
This issue has been through change control and I received approval to remove the topic last week.
I sent an email to aos-devel and openshift-sme on Firday, April 3. I can merge that PR to remove the nodes-nodes-audit-log-basic-config.adoc topic or close mine and go with this pr to pick up the other changes?
https://github.com/openshift/openshift-docs/pull/19721

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 1]
```diff
@@ -3,20 +3,20 @@
 // * nodes/nodes-audit-log.adoc
 
 [id="nodes-nodes-audit-log-basic-viewing_{context}"]
-= Viewing the API Audit Log
+= Viewing the audit log
 
-You can view the basic audit log.
+You can view logs for the master nodes for the {product-title} API server or the Kubernetes API server.
```
Comment (mburke5678) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:8]: @bergerhoffer This line is confusing me.  This makes it sound like you can view logs for each API server master node. 
Is it that the logs are from the API servers and you can view them on the OpenShift master?

File: modules/nodes-nodes-audit-log-basic.adoc [hunk 2]
```diff
@@ -3,30 +3,34 @@
...
-Example output for the Kubelet API server:
+= About the API audit log
+
+Audit works at the API server level, logging all requests coming to the server. Each audit log contains the following information:
+
+.Audit log fields
+[cols="1,2",options="header"]
+|===
+|Field |Description
+|`level` | The audit level at which the event was generated
```
Comment (mburke5678) [inline modules/nodes-nodes-audit-log-basic.adoc:14]: Period after `generated` for consistency.

File: modules/nodes-nodes-audit-log-basic.adoc [hunk 3]
```diff
@@ -3,30 +3,34 @@
...
+|`auditID` |A unique audit ID, generated for each request.
+|`stage` |The stage of the request handling when this event instance was generated.
+|`requestURI` |The request URI as sent by the client to a server.
+|`verb` |The Kubernetes verb associated with the request. For non-resource requests, this is the lowercase HTTP method.
+|`user` |The authenticated user information.
+|`impersonatedUser` |Optional. The impersonated user information.
```
Comment (mburke5678) [inline modules/nodes-nodes-audit-log-basic.adoc:20]: Do we need to define `impersonalted` user?

File: modules/nodes-nodes-audit-log-basic.adoc [hunk 4]
```diff
@@ -3,30 +3,34 @@
...
+|`sourceIPs` |Optional. The source IPs, from where the request originated and intermediate proxies.
```
Comment (mburke5678) [inline modules/nodes-nodes-audit-log-basic.adoc:21]: s/and intermediate/and any intermediate (or the intermediate)

File: modules/nodes-nodes-audit-log-basic.adoc [hunk 5]
```diff
@@ -3,30 +3,34 @@
...
+|`responseStatus` |Optional. The response status, populated even when the ResponseObject is not a Status type. For successful responses, this will only include the Code. For non-status type error responses, this will be auto-populated with the error Message.
+|`requestObject` |Optional. The API object from the request, in JSON format. The RequestObject is recorded as is in the request (possibly re-encoded as JSON), prior to version conversion, defaulting, admission or merging. It is an external versioned object type, and might not be a valid object on its own. This is omitted for non-resource requests and is only logged at Request Level and higher.
+|`responseObject` |Optional. The API object returned in the response, in JSON format. The ResponseObject is recorded after conversion to the external type, and serialized as JSON. This is omitted for non-resource requests and is only logged at Response Level.
+|`requestReceivedTimestamp` |The time that the request reached the API server.
+|`stageTimestamp` |The time that the request reached current audit stage.
```
Comment (mburke5678) [inline modules/nodes-nodes-audit-log-basic.adoc:28]: s/current/the current ??

**Comment by mburke5678** [comment]
@bergerhoffer A few nits and questions. Before merging we should chat about which PR we should use to remove the `nodes-nodes-audit-log-basic-config.adoc` topic (see my comment https://github.com/openshift/openshift-docs/pull/20905#issuecomment-609505285).

**Comment by bergerhoffer** [comment]
@vikram-redhat I stand corrected on this needing change control :). Thanks @mburke5678 for the information, and the review. I see you closed your PR in favor of this. I'll update per your feedback and then let's catch up on this tomorrow.

File: modules/nodes-nodes-audit-log-basic-viewing.adoc (hunk 1 above)
Comment (bergerhoffer) [reply modules/nodes-nodes-audit-log-basic-viewing.adoc:8]: @mburke5678 this wording was taken from another module in this PR, but I agree that it's confusing.

@soltysh Looks like each master has its own log, so would it be accurate to say something like `You can view logs for the {product-title} API server or the Kubernetes API server for each master node.` And @mburke5678 do you think this is any clearer?

**Comment by bergerhoffer** [comment]
@xingxingxia Can you please QE review this update? Thanks!

Preview: https://pr-20856-update--ocpdocs.netlify.com/openshift-enterprise/latest/nodes/nodes/nodes-nodes-audit-log.html#nodes-pods-audit-log-basic_nodes-nodes-audit-log

**Review by soltysh — APPROVED** [review]
lgtm from tech pov

**Comment by xingxingxia** [comment]
Will review today.

File: modules/nodes-nodes-audit-log-basic.adoc [hunk 6]
```diff
@@ -3,30 +3,34 @@
...
+= About the API audit log
+
+Audit works at the API server level, logging all requests coming to the server. Each audit log contains the following information:
+
+.Audit log fields
+[cols="1,2",options="header"]
+|===
+|Field |Description
+|`level` | The audit level at which the event was generated.
+|`auditID` |A unique audit ID, generated for each request.
```
Comment (xingxingxia) [inline modules/nodes-nodes-audit-log-basic.adoc:15]: A unique -> An unique

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 7]
```diff
@@ -3,20 +3,20 @@
...
+You can view logs for the {product-title} API server or the Kubernetes API server for each master node.
 
 .Procedure
 
-To view the basic audit log:
+To view the audit log:
 
 . View the {product-title} API server logs
 
 .. If necessary, get the node IP and name of the log you want to view:
```
Comment (xingxingxia) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:16]: > get the node IP and name
Should be only "get the node name", no IP.

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 8]
```diff
@@ -32,28 +32,28 @@ ip-10-0-170-165.ec2.internal audit.log
 .. View the {product-title} API server log for a specific master node and timestamp or view all the logs for that master:
 +
 ----
-$ oc adm node-logs <node-ip> <log-name> --path=openshift-apiserver/<log-name>
+$ oc adm node-logs <node-ip> --path=openshift-apiserver/<log-name>
```
Comment (xingxingxia) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:35]: Due to above comment, pls change the PR's all occurrences of `oc adm node-logs <node-ip>` to `oc adm node-logs <node-name>`

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 9]
```diff
@@ -32,28 +32,28 @@ ip-10-0-170-165.ec2.internal audit.log
 .. View the {product-title} API server log for a specific master node and timestamp or view all the logs for that master:
```
Comment (xingxingxia) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:32]: I'm confused what you mean by `log for a specific ... timestamp or view all the logs`? Do you mean there is a command `for a specific ... timestamp`, and there is a command `view all the logs`? If yes, could you give the command example respectively?

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 10]
```diff
@@ -32,28 +32,28 @@ ip-10-0-170-165.ec2.internal audit.log
...
...sponseComplete","requestURI":"/api/v1/namespaces/openshift-kube-controller-manager/configmaps/cert-recovery-controller-lock?timeout=35s","verb":"update","user":{"username":"system:serviceaccount:openshift-kube-controller-manager:localhost-recovery-client","uid":"dd4997e3-d565-4e37-80f8-7fc122ccd785","groups":["system:serviceaccounts","system:serviceaccounts:openshift-kube-controller-manager","system:authenticated"]},"sourceIPs":["::1"],"userAgent":"cluster-kube-controller-manager-operator/v0.0.0 (linux/amd64) kubernetes/$Format","objectRef":{"resource":"configmaps","namespace":"openshift-kube-controller-manager","name":"cert-recovery-controller-lock","uid":"5c57190b-6993-425d-8101-8337e48c7548","apiVersion":"v1","resourceVersion":"574307"},"responseStatus":{"metadata":{},"code":200},"requestReceivedTimestamp":"2020-04-02T08:27:20.200962Z","stageTimestamp":"2020-04-02T08:27:20.206710Z","annotations":{"authorization.k8s.io/decision":"allow","authorization.k8s.io/reason":"RBAC: allowed by ClusterRoleBinding \"system:openshift:operator:kube-controller-manager-recovery\" of ClusterRole \"cluster-admin\" to ServiceAccount \"localhost-recovery-client/openshift-kube-controller-manager\""}}
```
Comment (xingxingxia) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:48]: > The output appears similar to the following

The output does not have a preceding "ip-10-0-140-97.ec2.internal " in my test.

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 11]
```diff
@@ -32,28 +32,28 @@ ip-10-0-170-165.ec2.internal audit.log
...
 ----
 
 . View the Kubernetes API server logs:
 
 .. If necessary, get the node IP and name of the log you want to view:
```
Comment (xingxingxia) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:53]: Same, dhould be only "get the node name", no IP.

**Comment by xingxingxia** [comment]
LGTM except minor comment. CC'ing @wangke19 for knowing this feature.

File: modules/nodes-nodes-audit-log-basic.adoc (hunk 6 above)
Comment (bergerhoffer) [reply modules/nodes-nodes-audit-log-basic.adoc:15]: "A unique" is actually the correct usage.

**Comment by bergerhoffer** [comment]
Thanks @xingxingxia for the good catches, I've updated. Please let me know if you or @wangke19 have any more feedback or if this is good to go. Thanks!

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 12]
```diff
@@ -32,28 +32,28 @@ ip-10-0-170-165.ec2.internal audit.log
...
...sponseComplete","requestURI":"/api/v1/namespaces/openshift-kube-controller-manager/configmaps/cert-recovery-controller-lock?timeout=35s","verb":"update","user":{"username":"system:serviceaccount:openshift-kube-controller-manager:localhost-recovery-client","uid":"dd4997e3-d565-4e37-80f8-7fc122ccd785","groups":["system:serviceaccounts","system:serviceaccounts:openshift-kube-controller-manager","system:authenticated"]},"sourceIPs":["::1"],"userAgent":"cluster-kube-controller-manager-operator/v0.0.0 (linux/amd64) kubernetes/$Format","objectRef":{"resource":"configmaps","namespace":"openshift-kube-controller-manager","name":"cert-recovery-controller-lock","uid":"5c57190b-6993-425d-8101-8337e48c7548","apiVersion":"v1","resourceVersion":"574307"},"responseStatus":{"metadata":{},"code":200},"requestReceivedTimestamp":"2020-04-02T08:27:20.200962Z","stageTimestamp":"2020-04-02T08:27:20.206710Z","annotations":{"authorization.k8s.io/decision":"allow","authorization.k8s.io/reason":"RBAC: allowed by ClusterRoleBinding \"system:openshift:operator:kube-controller-manager-recovery\" of ClusterRole \"cluster-admin\" to ServiceAccount \"localhost-recovery-client/openshift-kube-controller-manager\""}}
```
Comment (wangke19) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc:48]: If the doc is for OCP 4.4, please remove the master node 'ip-10-0-140-97.ec2.internal ' from output, I don't see it in 4.4. Also apply this for below kube-apiserver related content.

Comment (bergerhoffer) [reply modules/nodes-nodes-audit-log-basic-viewing.adoc]: Confirmed on 4.4 that it doesn't show the node name, it just starts with `{"kind":"Event", ...`. Removed from the 3 example outputs.

@wangke19 Let me know if this looks good now, thanks!

**Comment by xingxingxia** [comment]
LGTM.

File: modules/nodes-nodes-audit-log-basic-viewing.adoc (hunk 12 above)
Comment (xingxingxia) [reply modules/nodes-nodes-audit-log-basic-viewing.adoc]: `oc adm node-logs <one single node name>` does not show the preceding node name.
`oc adm node-logs <node name 1> <node name 2> [<node name x> ...]` shows.
All right to do the removal per Ke Wang's advice.
Thanks!

**Comment by wangke19** [comment]
lgtm.

File: modules/nodes-nodes-audit-log-basic-viewing.adoc (hunk 12 above)
Comment (bergerhoffer) [reply modules/nodes-nodes-audit-log-basic-viewing.adoc]: Thanks for clarifying that!

**Pull request merged**